[Concepts] Changes that improve readability

bgilmer77 opened the pull request:
- Numerous changes to improve grammar and reader comprehension
- Addresses #4659

Signed-off-by:Brad Gilmer <brad@gilmer.tv>

bgilmer77 commented:
I am a `maintainer` over on the OH2 documentation team.  Am making this contribution with the hope that it helps new readers of the ESH documentation.  All comments are welcome and appreciated.

maggu2810 commented:
Thanks for your contribution.

We can check the ECA and fix your signed-off-by line if we merge this PR. No prob.
But perhaps you can use a correct Signed-off-by line next time yourself 😉 
The line looks like this one:

```text
Signed-off-by: Joe Somebody <somebody@someplace.net>
```

no github account or something else.

sjka commented on docs/documentation/concepts/index.md:
```diff
@@ -6,29 +6,36 @@ layout: documentation
-**Items** represent (fine-grained) functionality that is used by applications - as user interfaces or automation logic. Items have a state and they can receive commands.
+Things provide their functionality through a set of **Channels**.  Channels are "passive" and can be regarded as a declaration of a Thing; what it can offer.
+Whether an installation takes advantage of a particular capability depends on whether it has been configured to do so.
+SmartHome makes use of Channels through **Links** you establish to Items when you configure your system.
```
__Eclipse__ SmartHome [...]

bgilmer77 replied:
Agreed - this change will be in my next commit.

bgilmer77 replied:
Actually, I propose to drop ESH, since this does not make sense when the article is pulled into the OH documentation.  Leaving the reference in could confuse new users.

sjka commented on docs/documentation/concepts/index.md:
```diff
@@ -6,29 +6,36 @@ layout: documentation
+It is important to note that Things do not have to be physical devices; they can also represent a web service or any other manageable source of information and functionality.
 
-**Items** represent (fine-grained) functionality that is used by applications - as user interfaces or automation logic. Items have a state and they can receive commands.
+Things provide their functionality through a set of **Channels**.  Channels are "passive" and can be regarded as a declaration of a Thing; what it can offer.
```
I know, you haven't invented it, but the second sentence doesn't make any sense to me. Do you get what it should say? Should we drop it otherwise?

bgilmer77 replied:
Yes - I understand it, but it is not clear at all.  Also, the points it makes are important to computer programmers, not end users.  The text points out that Things are the things that have capabilities, and that these capabilities are accessed through Channels.  But the Channels themselves are not functions - they don't do anything.  They are simply the way a Thing expresses its capabilities.  As I said, users don't care about this.  I suggest replacing the sentence with the following:

"Things expose their capabilities through **Channels**."

sjka replied:
Much better IMHO 👍

bgilmer77 commented on docs/documentation/concepts/index.md:
```diff
@@ -6,29 +6,36 @@ layout: documentation
+Eclipse SmartHome is built around two distinctly different views.
+The first is the physical view - devices, configuration, connections, etc.
+The second is the functional view.
+This functional view consists of how information about the physical view is presented in User Interfaces, how automation operates on representations of physical devices, or how an action on a User Interface affects the software associated with the physical device it represents.
```
Agreed - this comment will be addressed in my next commit.

bgilmer77 commented on docs/documentation/concepts/index.md:
```diff
@@ -6,29 +6,36 @@ layout: documentation
-**Things** are the entities that can be physically added to a system and which can potentially provide many functionalities at once. It is important to note that things do not have to be devices, but they can also represent a web service or any other manageable source of information and functionality.
-Things provide their functionality through a set of **Channels**. Channels are "passive" and can be regarded as a declaration of a Thing, what it can offer. It is up to the individual setup, which of the Channels are actively used through Items (see below).
+**Things** are entities that can be physically added to a system.
+Things may provide more than one function (for example, a z-wave multi-sensor may provide a motion detector and also measure room temperature).
```
Agreed - this comment will be addressed in my next commit.

bgilmer77 commented:
@sjka, would it be okay to remove both references to the Eclipse SmartHome project?  Because this article also appears in the OH documentation, beginners may be confused by seeing references to ESH in the OH documentation.  I believe I can re-write the text in such a way that the information in the article is preserved.  Please advise.

kaikreuzer commented:
Although I am not @sjka, I would say: Yes please! We try to keep all documentation as re-usable as possible, so if you can come up with a good alternative formulation, please go ahead!

bgilmer77 commented:
@kaikreuzer well, I hope I have not gone too far Kai.  I have removed the references to ESH, but I have also made some changes that I hope make the article easier for a noob reader to understand.

@sjka, I believe I have addressed all of your comments, but perhaps by making more changes than necessary.  It is my hope that these changes help people who are just coming to the project for the first time.  I would appreciate your feedback.

bgilmer77 commented:
@maggu2810 @sjka Could you please review my latest changes and if they are okay, merge this PR?  These changes help those just getting familiar with ESH and OH.  With the new release coming, it would be good to have these in the docs.

Thank you.

maggu2810 commented on docs/documentation/concepts/index.md:
```diff
@@ -6,29 +6,43 @@ layout: documentation
+The functional view might be a new concept for you.
+The functional view focuses on how information about the devices, connections, and so on, is represented in user interfaces.
+The functional view includes focusing on how rules effect representations of physical devices in software.
+ Perhaps most important to you, the functional view focuses on how an action in a user interface affects the software associated with the physical device it represents.
```
Does this start with an whitespace by intention?

bgilmer77 replied:
It is intended to be the start of a new paragraph focusing on the functional view.  The physical view was the focus of the previous paragraph.  Is that what you are seeing?  I get a failure when I try to render the page using jekyll, so I cannot see how the HTML looks.

maggu2810 commented on docs/documentation/concepts/index.md:
```diff
@@ -6,29 +6,43 @@ layout: documentation
 
+The glue between Things and Items are **Links**.
+A Link is an association between exactly one Channel and one Item.
+If a Channel is linked to an Item, it is "enabled", which means that the capability the Item represents is accessible through that Channel.
```
Hm, the Channel does not access the Item. Or do he?

bgilmer77 replied:
When I did the edit, I tried to clarify the original wording without changing the meaning.  

What I see here is that a Thing can have multiple Capabilities that are exposed through Channels.  If you want to get at a Capability of a Thing, then you have to Link that Channel of that Thing to an Item.  Once you have done that, you can control that Capability of that Thing using the linked Item.  Isn't that correct?  Do you have a different wording suggestion?

maggu2810 commented on docs/documentation/concepts/index.md:
```diff
@@ -6,29 +6,43 @@ layout: documentation
+The actuator is a Thing that might be installed in an electrical cabinet.
+It has a physical address and it must be configured in order to be used (remember the physical view introduced at the beginning of this article).
 
+In order for the user to control the two lights, he or she access the capability of the actuator Thing (turning on and off two separate lights) through two Channels, that are Linked to two switch Items presented to the user through a user interface.
```
Couldn't be "access the capability of the actuator Thing through two Channels, that are Linked to two switch Items" interpret that he still accesses the channels?

bgilmer77 replied:
I don't think so.  He access the Capability of the Thing **through** the Channel.  Does that help at all?

bgilmer77 commented:
@maggu2810 the word 'Capability' is defined in the Oxford English Dictionary as, "The power or ability to do something".  I believe this is exactly what is intended in this article, so I believe the word is correct, but I am open to any other suggestions.

I am involved in another GitHub project, [AMWA NMOS](https://www.github.com/AMWA-TV/nmos).  In this project, conceptually we talk about a Platform offering up a Capability through a microService to an Application, which consumes (or makes use of) this Capability in order to accomplish a task for an end user.  I do not believe the idea of Capability is widely used in ESH or OH, but it could be a very useful concept.  @kaikreuzer , for your consideration.

bgilmer77 commented:
FYI, [here](http://mdc.posick.net/2013/12/capability-based-programming.html) is a link to a document on Capability-based Programming...

bgilmer77 commented:
@maggu2810, @sjka, I believe this PR is ready for merging if you are okay with using the term "Capability".  Please see discussion in PR above.  If not, could you please let me know so I can change the wording?  Over on the OH side we have other suggestions to make to this article and I do not want to end up having to manually merge later.  Thank you.

maggu2810 commented:
No veto from my side :wink:

bgilmer77 commented:
@sjka & @maggu2810, thanks, and Happy New Year!
